libmscore/read206.cpp

```diff
@@ -1418,13 +1418,14 @@ static bool readTextPropertyStyle206(XmlReader& e, TextBase* t, Element* be)
                               return;
                               }
                         else {
+                              // Please forgive me for this.
                               // Check for start tag
-                              QRegExp startRe("<[A-z0-9]+>");
+                              QRegExp startRe("<\\w+((\\s+\\w+(\\s*=\\s*(?:\".*?\"|'.*?'|[\\^'\">\\s]+))?)+\\s*|\\s*)/?>");
```

Comment:
Do we really need all this? :) Our goal here is to detect open and close tags presence rather than verify syntax correctness of the read files.

Anyway, there is another issue with these expressions. The given regex for the start tag matches also self-closing tags like `<tag attr="val"/>` (see `/?` part in the end). The regex for the end tag does not match it (since it has the requirement for `/` character in the beginning). That means that when this procedure meets such a tag the `depth` counter will be incremented rather than left at the same value. The concern is more than real: `pos` or `offset` tags are exactly self-closing tags there and it is very probable to meet them somewhere inside an element.

So I would propose something of these options:
1) Use some simpler regular expression (something like `<[^>]*[^/>]>`?).
2) Search for the fixed end tag with the given name like I did it for `SystemText` case:
https://github.com/musescore/MuseScore/blob/c14e51f41ce576e302b195b297e76660dc190d5c/libmscore/read206.cpp#L3014
This should be enough if you wish to read just within the score of the tag which have just opened but you should be sure then to call this readahead procedure only at the beginning of the tag scope.

Of course, it is better to avoid such tricks at all but here this doesn't seem to be feasible quick enough.

Reply:
For 1. the problem is that if you have an xml attribute like `text="something"`, it could be `text="here >"` which would trick the regex.

For 2. the problem is that the closing tag can be anything - it won't just be `</Text>`.

So that's why this 'depth' system is needed, which is for working out when we've come out of the main tag that we started within.

This regex does have a problem in that it doesn't work :)
I'll find a better solution.

Reply:
Is including `>` to xml attributes values really allowed?

For the second approach, you can obtain the actual name of the tag to search via `e.name()` call to `XmlReader` (again, if you are sure you are at the beginning of the scope). Counting depth is indeed more universal considering this question.

Reply:
1. Yes, ">" is allowed in string values for xml attributes.

2. That might have to be the way to do it :/ At least I do know that when this is called, it is at the beginning of the scope.

Reply:
Should be better now @dmitrio95